pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/VarByteSingleValueReader.java

```diff
@@ -0,0 +1,166 @@
+   */
+  public VarByteSingleValueReader(PinotDataBuffer pinotDataBuffer, ChunkUncompressor uncompressor)
+      throws IOException {
+    _chunkUncompressor = uncompressor;
+    _dataBuffer = pinotDataBuffer;
+
+    int headerOffset = INT_SIZE; // First entry is the version, which is unused currently.
+    _numChunks = _dataBuffer.getInt(headerOffset);
+    headerOffset += INT_SIZE;
+
+    _numDocsPerChunk = _dataBuffer.getInt(headerOffset);
+    headerOffset += INT_SIZE;
+
+    _lengthOfLongestEntry = _dataBuffer.getInt(headerOffset);
+    headerOffset += INT_SIZE;
+
+    int chunkHeaderSize = _numDocsPerChunk * INT_SIZE;
+    _maxChunkSize = chunkHeaderSize + (_lengthOfLongestEntry * _numDocsPerChunk);
+
+    // Slice out the header from the data buffer.
+    int headerLength = _numChunks * INT_SIZE;
+    _header = _dataBuffer.view(headerOffset, headerOffset + headerLength);
+  }
+
+  @Override
+  public String getString(int row, VarByteReaderContext context)
+      throws IOException {
+
+    int chunkRowId = row % _numDocsPerChunk;
+    ByteBuffer chunkBuffer = getChunkForRow(row, context);
+
+    int rowOffset = chunkBuffer.getInt(chunkRowId * INT_SIZE);
+    int nextRowOffset = chunkBuffer.getInt((chunkRowId + 1) * INT_SIZE);
+
+    // For incomplete chunks, the next string's offset will be 0 as row offset for absent rows are 0.
+    if ((chunkRowId == (_numDocsPerChunk - 1)) || (nextRowOffset == 0)) {
+      nextRowOffset = chunkBuffer.limit();
+    }
+
+    int length = nextRowOffset - rowOffset;
+    chunkBuffer.position(rowOffset);
+
+    // Lazy allocation of thread local memory.
+    byte[] bytes = _reusableBytes.get();
+    if (bytes == null) {
+      bytes = new byte[_lengthOfLongestEntry];
+      _reusableBytes.set(bytes);
+    }
+
+    chunkBuffer.get(bytes, 0, length);
+    return new String(bytes, 0, length, UTF_8);
+  }
+
+  @Override
+  public void close() {
+    // Nothing to close here.
+  }
+
+  @Override
+  public VarByteReaderContext createContext() {
+    return new VarByteReaderContext(_maxChunkSize);
+  }
+
+  /**
+   * Helper method to get the chunk for a given row.
+   * <ul>
+   *   <li> If the chunk already exists in the reader context, returns the same. </li>
+   *   <li> Otherwise, loads the chunk for the row, and sets it in the reader context. </li>
+   * </ul>
+   * @param row Row for which to get the chunk
+   * @param context Reader context
+   * @return Chunk for the row
+   */
+  private ByteBuffer getChunkForRow(int row, VarByteReaderContext context)
+      throws IOException {
+    int chunkId = row / _numDocsPerChunk;
+    if (context.getChunkId() == chunkId) {
+      return context.getChunkBuffer();
+    }
+
+    int chunkSize;
+    int chunkPosition = getChunkPosition(chunkId);
+
+    // Size of chunk can be determined using next chunks offset, or end of data buffer for last chunk.
+    if (chunkId == (_numChunks - 1)) { // Last chunk.
+      chunkSize = (int) (_dataBuffer.size() - chunkPosition);
```

Comment:
Yeah, some members have showed preference on if/else over ?:. I guess we can't make everyone happy :-)